spec/operators/catch-spec.ts

```diff
@@ -278,4 +279,84 @@ describe('Observable.prototype.catch', () => {
       done();
     });
   });
+
+  context('fromPromise', () => {
+    type SetTimeout = (callback: (...args: any[]) => void, ms: number, ...args: any[]) => NodeJS.Timer;
+
+    let trueSetTimeout: SetTimeout;
+    let sandbox: sinon.SinonSandbox;
+    let timers: sinon.SinonFakeTimers;
+
+    let source: Rx.Observable<any>;
+    beforeEach(() => {
```

Comment:
we have few cases of using it, but in general if it can be avoided try not to use it. except some obvious cases (like test scheduler), usually test for Rx can be written without sandbox.